tests/VirtoCommerce.NotificationsModule.Tests/UnitTests/NotificationMessageSearchServiceUnitTests.cs

```diff
@@ -0,0 +1,211 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using MockQueryable;
+using Moq;
+using VirtoCommerce.NotificationsModule.Core.Model;
+using VirtoCommerce.NotificationsModule.Core.Model.Search;
+using VirtoCommerce.NotificationsModule.Data.Model;
+using VirtoCommerce.NotificationsModule.Data.Repositories;
+using VirtoCommerce.NotificationsModule.Data.Services;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Core.Events;
+using Xunit;
+
+namespace VirtoCommerce.NotificationsModule.Tests.UnitTests;
+
+public class NotificationMessageSearchServiceUnitTests
+{
+    private readonly Mock<INotificationRepository> _repositoryMock;
+    private readonly NotificationMessageSearchService _searchService;
+
+    public NotificationMessageSearchServiceUnitTests()
+    {
+        AbstractTypeFactory<NotificationMessage>.RegisterType<EmailNotificationMessage>();
+
+        _repositoryMock = new Mock<INotificationRepository>();
+        var messageService = new NotificationMessageService(() => _repositoryMock.Object, new Mock<IEventPublisher>().Object);
+        _searchService = new NotificationMessageSearchService(() => _repositoryMock.Object, messageService);
+    }
+
+    [Fact]
+    public async Task SearchMessageAsync_ShouldFilterByTenantIdAndType()
+    {
+        // Arrange
+        var tenantId = "tenantId1";
+        var tenantType = "Store";
+
+        var entities = new List<NotificationMessageEntity>
+        {
+            new EmailNotificationMessageEntity
+            {
+                Id = Guid.NewGuid().ToString(),
+                TenantId = tenantId,
+                TenantType = tenantType,
+            }
+        };
```

Comment:
Need to add more entities to each filtering test to make sure the filtering actually works.
Now we have one input entity and one ouput entity - this doesn't show any filtering.